cmd/ewasm/main.go

```diff
@@ -54,11 +54,38 @@ func runCmd(ctx *cli.Context) error {
 				return errors.New(fmt.Sprintf("Could not read module: %v", err))
 			}
 
-			_, err = exec.NewVM(m)
+			vm, err := exec.NewVM(m)
 			if err != nil {
 				return errors.New(fmt.Sprintf("Could not instantiate vm: %v", err))
 			}
 
+			// If the module doesn't have any start function, check if the
+			// exports specify a `main` function and if so, execute it.
```

Comment:
Also if the module has a start function, that shouldn't be executed as per current "ECI rules".